moto/awslambda/models.py

```diff
@@ -595,23 +618,42 @@ def get_configuration(self):
             "StateTransitionReason": "User initiated",
         }
 
+    def delete(self, region_name):
+        lambda_backend = lambda_backends[region_name]
+        lambda_backend.delete_event_source_mapping(self.uuid)
+
     @classmethod
     def create_from_cloudformation_json(
         cls, resource_name, cloudformation_json, region_name
     ):
         properties = cloudformation_json["Properties"]
-        func = lambda_backends[region_name].get_function(properties["FunctionName"])
-        spec = {
-            "FunctionArn": func.function_arn,
-            "EventSourceArn": properties["EventSourceArn"],
-            "StartingPosition": properties["StartingPosition"],
-            "BatchSize": properties.get("BatchSize", 100),
-        }
-        optional_properties = "BatchSize Enabled StartingPositionTimestamp".split()
-        for prop in optional_properties:
-            if prop in properties:
-                spec[prop] = properties[prop]
-        return EventSourceMapping(spec)
+        lambda_backend = lambda_backends[region_name]
+        return lambda_backend.create_event_source_mapping(properties)
+
+    @classmethod
+    def update_from_cloudformation_json(
+        cls, new_resource_name, cloudformation_json, original_resource, region_name
+    ):
+        properties = cloudformation_json["Properties"]
+        event_source_uuid = original_resource.uuid
+        lambda_backend = lambda_backends[region_name]
+        return lambda_backend.update_event_source_mapping(event_source_uuid, properties)
+
+    @classmethod
+    def delete_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        properties = cloudformation_json["Properties"]
+        lambda_backend = lambda_backends[region_name]
+        esms = lambda_backend.list_event_source_mappings(
+            event_source_arn=properties["EventSourceArn"],
+            function_name=properties["FunctionName"],
+        )
+
+        for esm in esms:
+            if esm.logical_resource_id in resource_name:
+                lambda_backend.delete_event_source_mapping
+                esm.delete(region_name)
```

Comment:
I'm guessing you want to delete the `resource` here, not the region :)

Reply:
Maybe I missunderstood but this deletes the resource. The problem here is doing this "crazy" `in` condition veryfing if it's possible to delete based on names.

Reply:
Yup, gotcha. I should've checked the signature of the delete-method - I assumed you wanted to write `esm.delete(resource_name)`. All good.